### PR TITLE
Exodus Fixs.

### PR DESCRIPTION
**Pull request by Argalep**
* Exodus Dungeon mobiles fix. (do not autodispel, stat fixs)
* Exodus Chest drop item fix.
* Exodus Chest using the Detect Hidden skill, It will be deleted after 5 minutes
* Exodus Chest Spawns fix. 
* Exodus Chest found  min. 98 detect hidden
* If the weight of the contest is 0, do not show its weight

ref: 

- https://uo.com/wiki/ultima-online-wiki/combat/pvm-player-versus-monster/stealing-in-exodus-dungeon/

- http://www.uoguide.com/The_Exodus_Encounter

### CODE REVIEW

**Comment by dmurphy22** [comment]
Thank you for your contribution, can you please add a description of this PR as per the thread  https://www.servuo.com/tutorials/how-to-make-a-good-pull-request-at-github.33/

File: Scripts/Services/Revamped Dungeons/TheExodusEncounter/Items/ExodusChest.cs [hunk 1]
```diff
@@ -1,70 +1,203 @@
...
+        public override int LabelNumber { get { return 1073403; } } // decorative box
+
+        public static Type[] RituelItem { get { return m_RituelItem; } }
+
+        private static readonly Type[] m_RituelItem = new[]
+        {
+            typeof(ExodusSummoningRite), typeof(ExodusSacrificalDagger), typeof(RobeofRite), typeof(ExodusSummoningAlter)
+        };
+
+        private static Type[] m_ImbuingEssenceIngreds =
```
Comment (ppardalj) [inline Scripts/Services/Revamped Dungeons/TheExodusEncounter/Items/ExodusChest.cs]: This code is pretty similar to the one in `TreasureMapChest` and `EssenceBox` classes. I would propose to extract these definitions to the `Loot` class along with other loot constructors, so you just need to call:

```
var essence = Loot.RandomEssence();
DropItem(essence);
```

You can extract the Exodus ritual items as well in a similar way.

File: Scripts/Skills/DetectHidden.cs [hunk 2]
```diff
@@ -80,6 +81,20 @@ protected override void OnTarget(Mobile src, object targ)
                     }
 
                     inRange.Free();
+                    
+                    IPooledEnumerable ExodusChestInRange = src.Map.GetItemsInRange(p, 3);
```
Comment (ppardalj) [inline Scripts/Skills/DetectHidden.cs]: You can take advantage of LINQ and foreach implicit disposal to clean up this code:

```
foreach (var chest in src.Map.GetItemsInRange(p, 3).OfType<ExodusChest>().Where(it => !it.Visible))
{
	chest.Visible = true;
	chest.StartDeleteTimer();
}
```

File: Scripts/Services/Revamped Dungeons/TheExodusEncounter/Items/ExodusChest.cs [hunk 3]
```diff
@@ -1,70 +1,203 @@
...
 
 namespace Server.Items
 {
     public class ExodusChest : LockableContainer
     {
+        public override int LabelNumber { get { return 1073403; } } // decorative box
+
+        public static Type[] RituelItem { get { return m_RituelItem; } }
+
+        private static readonly Type[] m_RituelItem = new[]
```
Comment (ppardalj) [inline Scripts/Services/Revamped Dungeons/TheExodusEncounter/Items/ExodusChest.cs]: Is this a typo? Prefer `RitualItem` over `RituelItem`

**Pull request merged**